datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2114,7 +2116,9 @@ pub struct Values {
 // Manual implementation needed because of `schema` field. Comparison excludes this field.
 impl PartialOrd for Values {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-        self.values.partial_cmp(&other.values)
+        self.values
+            .partial_cmp(&other.values)
+            .filter(|cmp| *cmp != Ordering::Equal || self == other)
```

Comment:
I think `self == other` will (re) compare all entries in `self.values` and `other.values` again

we can probably avoid checking `values == other.values` here by just checking the missing field

```suggestion
            .filter(|cmp| *cmp != Ordering::Equal || self.schema == other.schema)
```

Reply:
The same comment applies ot the other comparisons in this PR as well

Reply:
You're right, but the proposed code is not self-maintainable. I.e. the code likely won't be updated when (a) a new field gets added and (b) neither compiler nor any tests complains.
For this to work, we would need a bigger change for the Ord implementation where we destructure the compared object, which then forces code update when any new field is added.

I'd propose this more elaborate change to be a follow-up, if you don't mind?